pip/req/req_set.py

```diff
@@ -570,13 +602,32 @@ def _pip_has_created_build_dir(self):
             )
         )
 
+    def _to_install(self):
+        """Create the installation order.
+
+        We install the user specified things in the order given, except when
+        dependencies require putting other things first.
```

Comment:
So how about I say its undefined except that we deploy deps before depending, with arbitrary cycle breaking?